src/target/riscv/riscv-013.c

```diff
@@ -4365,17 +4358,16 @@ read_memory_progbuf(struct target *target, const riscv_mem_access_args_t args)
 
 	uint64_t mstatus = 0;
 	uint64_t mstatus_old = 0;
-	if (modify_privilege(target, &mstatus, &mstatus_old) != ERROR_OK)
+	uint64_t dcsr = 0;
```

Comment:
Please, use `riscv_reg_t` instead of `uint64_t` for `mstatus` and `dcsr` values.

Reply:
address